src/librustc_metadata/cstore.rs

```diff
@@ -218,6 +222,23 @@ impl CStore {
         self.used_libraries.borrow_mut().push(lib);
     }
 
+    // Update kind and, optionally, the name of all native libaries (there may be more than one)
+    // with the specified name.
+    pub fn update_used_library(&self, name: &str, new_name: Option<&str>,
+                               new_kind: NativeLibraryKind) -> bool {
+        let mut found = false;
+        for item in self.used_libraries.borrow_mut().iter_mut() {
+            if item.name == name {
+                item.kind = new_kind;
+                if let Some(new_name) = new_name {
+                    item.name = Symbol::intern(new_name);
```

Comment:
Hm that does sound like an unfortunate "regression", but I'd imagine that in practice that was doomed to never link correctly anyway?

It does seem like we can't strictly require that `-lfoo` isn't specified more than once, but perhaps we can still be strict about *renamings* where a lib is only renamed once?